utils/puppeteerScrapper.js

```diff
@@ -1,151 +1,90 @@
 const puppeteer = require('puppeteer'); // v13.0.0 or later
 
-const getWaifu = async () => {
-  const waifu = {
-    name: '',
-    imageUrl: '',
-    bioUrl: '',
-    extract: '',
-    appearsIn: '',
+const nameSelector = 'h1';
+const imageSelector = '#widget-waifu-of-the-day > a > img';
+const popularitySelector = '#popularity-rank';
+const ageSelector = '#age';
+const husbandoSelector = '#waifu-classification';
+const appearsSelector = '#waifu-core-information > div > div > a';
+
+const waifuURL = "https://mywaifulist.moe/"
+const randomUrl = "https://mywaifulist.moe/random"
+
+const getWaifu = async (retry = false) => {
+  const [browser, page] = await intializeScrapper()
+  await page.setViewport({ "width": 1920, "height": 1080 })
+
+  const url = retry ? randomUrl : waifuURL
+  console.info(`Navigating to: ${url}`)
+  await page.goto(url, {waitUntil: "networkidle2"});
+
+  if (!retry) {
+    console.info("Looking for WOTD section...");
+    const section = await page.waitForSelector(imageSelector)
+
+    console.info("Loading waifu info");
+    await section.click()
+    await page.waitForNavigation({waitUntil: "networkidle2"})
   }
 
+  console.info("Detecting Husbandos...");
+  const isHusbando = await page.$(husbandoSelector)
 
-  const hrefSelector = '#widget-waifu-of-the-day > a';
-  const nameSelector = '#widget-waifu-of-the-day .text-zinc-100';
-  const imageSelector = '#widget-waifu-of-the-day > a > img';
-  const descriptionSelector = '#description';
-  const appearsSelector = '#waifu-core-information > div > div > a';
+  if (isHusbando) {
+    console.info("Husbando detected, retry a random waifu...")
+    await browser.close();
+    return await getWaifu(true)
+  }
```

Comment:
I don't see the need of having this validation. We can create a PR to add this validation if the counsel decides to remove all husbandos from the list.